pkg/handler/eventhandler.go

```diff
@@ -201,13 +189,28 @@ func WithLowPriorityWhenUnchanged[object client.Object, request comparable](u Ty
 	}
 }
 
-type workqueueWithCustomAddFunc[request comparable] struct {
-	workqueue.TypedRateLimitingInterface[request]
-	addFunc func(item request, q workqueue.TypedRateLimitingInterface[request])
+type workqueueWithDefaultPriority[request comparable] struct {
+	priorityqueue.PriorityQueue[request]
+	priority int
```

Comment:
I'm wondering if we should also make this a pointer

So that we only call AddWithOpts with a priority if it is actually set (and not &0 if unset)

I know that **currently** for **our** priority queue implementation it doesn't matter (as far as I can tell), still wondering if it would be better